bokehjs/src/coffee/renderer/annotation/tooltip.coffee

```diff
@@ -54,17 +54,17 @@ define [
       @$el.removeClass('bk-right')
       @$el.removeClass('bk-left')
 
-      if side == "right"
-        @$el.addClass("bk-left")
-        top  = sy - @$el.height()/2
-        left = sx + 18
-      else if side == "left"
-        @$el.addClass("bk-right")
-        top  = sy - @$el.height()/2
-        left = sx - @$el.width() - 23
-      else
-        logger.warn("invalid tooltip side: '#{side}'")
-        return
+      arrow_width = 10
+
+      switch side
+        when "right"
+          @$el.addClass("bk-left")
+          left = sx + (@$el.outerWidth() - @$el.innerWidth()) + arrow_width
+        when "left"
+          @$el.addClass("bk-right")
+          left = sx - @$el.outerWidth() - arrow_width
+
+      top = sy - @$el.outerHeight()/2
```

Comment:
why removing else case to handle unexpected side values? Or is this validation being done elsewhere?